src/sentry/integrations/bitbucket/integration.py

```diff
@@ -96,11 +96,27 @@ def get_repositories(self, query=None):
                 for repo in resp.get("values", [])
             ]
 
-        full_query = (u'name~"%s"' % (query)).encode("utf-8")
-        resp = self.get_client().search_repositories(self.username, full_query)
-        return [
-            {"identifier": i["full_name"], "name": i["full_name"]} for i in resp.get("values", [])
-        ]
+        exact_query = (u'name="%s"' % (query)).encode("utf-8")
+        fuzzy_query = (u'name~"%s"' % (query)).encode("utf-8")
+
+        exact_search_resp = self.get_client().search_repositories(self.username, exact_query)
+        fuzzy_search_resp = self.get_client().search_repositories(self.username, fuzzy_query)
+
+        if exact_search_resp.get("values", []) == []:
```

Comment:
Can we simplify this by doing `len(exact_search_resp.get("values")) == 0`? Will probably need to check the response from bitbucket when values is empty.

Reply:
@manuzope Do you mean check status code?